core/src/com/unciv/ui/tilegroups/TileGroup.kt

```diff
@@ -299,17 +301,17 @@ open class TileGroup(
         }
         tileImageIdentifiers = tileBaseImageLocations
 
-        for (image in tileBaseImages.asReversed()) image.remove()
+        for (image in tileBaseImages) image.remove()
         tileBaseImages.clear()
-        for (baseLocation in tileBaseImageLocations.asReversed()) { // reversed because we send each one to back
```

Comment:
I've removed both reverse calls as they were just unnecessary.
The first one: why reverse a list to remove every image from it?
The second one: Instead of reversing the list and adding everything at the front (line 327 (removed)), everything is just added in order now